How-To/Create-a-New-Residential-Feature.md

```diff
@@ -0,0 +1,177 @@
+
+-   Upload the cleaned version to `ccao-data-warehouse-us-east-1` as a parquet file, grouping the file by year.
+
+------------------------------------------------------------------------
+
+## Step 3: [Add a Model to the dbt DAG](https://github.com/ccao-data/data-architecture/blob/master/dbt/README.md#-how-to-add-a-new-model) to Transform the Data into a [Model View](https://github.com/ccao-data/data-architecture/tree/master/dbt/models).
+
+-   Once data has been cleaned and is ready for processing, it can be can be written using a SQL query, or as a Python script using Athena PySpark's [built-in third-party packages](https://docs.aws.amazon.com/athena/latest/ug/notebooks-spark-preinstalled-python-libraries.html). A list of commonly used queries exist in [dbt-macros](https://github.com/ccao-data/data-architecture/tree/master/dbt/macros). The most commonly used macros involve spatial transformations, such as identifying the distance to nearest geography of a particular type (i.e. stadiums).
+
+-   At the top of your script, make sure that the parquet outputs are partitioned by year. For SQL queries these are done with the following structure:
+
+```         
+{{config(          
+materialized='table',         
+partitioned_by=['year'],          
+bucketed_by=['pin10'],          
+bucket_count=1)}
+```
+
+If the output does not have the partition column as the final column, you will receive the following error:
+
+```         
+HIVE_COLUMN_ORDER_MISMATCH: Partition keys must be the last columns in the table and in the same order as the table properties: 
+```
+
+-   Build out the transformation. To ensure consistency, use one of the macros whenever possible, each of which utilizes a specific coding structure. For example, the `dist_to_nearest_geometry.sql` requires a value for `pin`, `year`, `x_3435`, and `y_3435`. These also need to be structured in accordance with specific data types. In this case, if geographies are classified as a string, you will run into the following error.
+
+```         
+Runtime Error in model spatial.spatial_stadium (models/spatial/spatial.spatial_stadium.sql) line 9:5: Unknown type: STRING
+```
+
+To remedy this, you can recast the geometry value as a binary points.
+
+```         
+ST_ASBINARY(ST_POINT(stadium.lon, stadium.lat)) AS geometry
+```
+
+-   Once the macro / sql query is correct, run it in your terminal to construct the output in your development database of `AWS Athena`. The output view is the file which will be referenced during `Step 4`. A sample build is presented below.
+
+```         
+dbt build --select proximity.dist_pin_to_stadium
+```
+
+-   Complete the documentation by configuring it as a [source](https://docs.getdbt.com/docs/build/sources) in the folder's dbt `schema.yml` file, and update the folder's `docs.md` and `columns.md` files.
```

Comment:
**[Nitpick, non-blocking]** I believe the table created by this step should be configured as a model, not a source -- the raw data that is used in the transformation (either from `ccao-data-raw-us-east-1` or `ccao-data-warehouse-us-east-1`) but whose extract/transform script is defined elsewhere should be a source, but any transformations defined in dbt should be models.

Reply:
@Damonamajor Looks like this suggestion still needs to be incorporated!